cmd/rollkit/commands/intercept.go

```diff
@@ -0,0 +1,125 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	cometos "github.com/cometbft/cometbft/libs/os"
+	"github.com/spf13/cobra"
+
+	rollconf "github.com/rollkit/rollkit/config"
+)
+
+const rollupBinEntrypoint = "entrypoint"
+
+var (
+	rollkitConfig rollconf.TomlConfig
+
+	// ErrRollkitCommand is returned when the command is a rollkit command
+	// which we don't want to intercept.
+	ErrRollkitCommand = fmt.Errorf("rollkit command")
+
+	// ErrHelpVersion is returned when the user attempts to run help or version.
+	ErrHelpVersion = fmt.Errorf("help or version")
+
+	// ErrRunEntrypoint is returned when the entrypoint fails to run.
+	ErrRunEntrypoint = fmt.Errorf("run rollup entrypoint")
```

Comment:
these are very useful errors. 
like what am I supposed to do when I see `rollkit command` as the error? 

Looking at `InterceptCommand`, these are normal use cases and you are forcing yourself to return an error that isn't an error. 

Instead I would update the `InterceptCommand` to return a boolean and error. 
```go
func InterceptCommand() (executed bool, err error)
```

Because there ultimately can be 4 different scenarios. 

1. Valid path of executing a command without error
2. Valid path of executing a command, but encounters an error
3. No command should be executed, no error encounter
4. Unsure if/no command should be executed, error encounted

Then where we are returning `ErrRollkitCommand` and `ErrHelpVersion` it should just return `false, nil` and the main cmd should continue executing.

Reply:
They are not a real errors, but rather used for wrapping as prefixes and to keep main function logic clean and easy to understand what happened.

Would be better to have just a bool flag and error if any, yes